### PR TITLE
[AUTO] Disable CPU model cache for fallback

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Pass empty cache_dir to CPU plugin if cache_dir is set by user app when enable_startup_fallback or enable_runtime_fallback is true

### Tickets:
 - 141026

### CODE REVIEW

File: src/plugins/auto/src/auto_schedule.cpp [hunk 1]
```diff
@@ -174,6 +175,11 @@ void AutoSchedule::init() {
                     m_compile_context[CPU].m_device_info = *cpu_iter;
                     m_compile_context[CPU].m_device_info.config[ov::hint::performance_mode.name()] =
                         ov::hint::PerformanceMode::LATENCY;
+                    if (m_compile_context[ACTUALDEVICE].m_device_info.config.count(ov::cache_dir.name()) &&
+                        (m_context->m_startup_fallback || m_context->m_runtime_fallback)) {
+                        m_compile_context[CPU].m_device_info.config[ov::cache_dir.name()] = "";
+                        LOG_INFO_TAG("Clear cache dir setting for CPU accelerator");
+                    }
```
Comment (songbell) [inline src/plugins/auto/src/auto_schedule.cpp:182]: is it possible to move all cache related customization/operations to one function/one single entry? 
we have cache operations a few lines above this change, and then some other cache operation here....

Comment (yangwang201911) [reply src/plugins/auto/src/auto_schedule.cpp:182]: will implement.

Comment (yangwang201911) [reply src/plugins/auto/src/auto_schedule.cpp:173]: Updated.

File: src/plugins/auto/src/auto_schedule.cpp [hunk 2]
```diff
@@ -130,6 +130,53 @@ void AutoSchedule::init() {
             m_firstload_promise.set_value();
         });
     };
+    auto update_cache_setting = [this](bool is_actual_cpu,
```
Comment (songbell) [inline src/plugins/auto/src/auto_schedule.cpp:133]:
```suggestion
    auto customize_context_from_cache_setting = [this](bool is_actual_cpu,
```

**Review by songbell — APPROVED** [review]

**Review by zhaixuejun1993 — APPROVED** [review]

**Pull request merged**